Tasks/DotNetCoreCLIV2/dotnetcore.ts

```diff
@@ -325,22 +326,50 @@ export class dotNetExe {
         }
 
         var projectFiles = utility.getProjectFiles(projectPattern);
+        var resolvedProjectFiles: string[] = [];
 
         if (searchWebProjects) {
-            projectFiles = projectFiles.filter(function (file, index, files): boolean {
+            resolvedProjectFiles = projectFiles.filter(function (file, index, files): boolean {
                 var directory = path.dirname(file);
                 return tl.exist(path.join(directory, "web.config"))
                     || tl.exist(path.join(directory, "wwwroot"));
             });
 
-            if (!projectFiles.length) {
-                tl.error(tl.loc("noWebProjectFound"));
-            }
+            if (!resolvedProjectFiles.length) {
+                var projectFilesUsingWebSdk = projectFiles.filter(this.isWebSdkUsed);
+                if(!projectFilesUsingWebSdk.length) {
+                    tl.error(tl.loc("noWebProjectFound"));
+                }
+                return projectFilesUsingWebSdk;
+            } 
+            return resolvedProjectFiles;
         }
-
         return projectFiles;
     }
 
+    private isWebSdkUsed(projectfile: string): boolean {
+        if (projectfile.endsWith('.vbproj')) return false
+
+        try {
+            var fileBuffer: Buffer = fs.readFileSync(projectfile);
+            var webConfigContent: string;
+            
+            var fileEncodings = ['utf8', 'utf16le'];
+
+            for(var i = 0; i < fileEncodings.length; i++) {
+                tl.debug("Trying to decode with " + fileEncodings[i]);
+                webConfigContent = fileBuffer.toString(fileEncodings[i]);
```

Comment:
Should this statement be also in the try block below

Reply:
It need not be, as this statement will not throw an error unless fileEncoding is not accepted value. Since fileEncoding is not user input, we could find bug during development easily, if we are to add a new encoding, if we don't wrap it around try-catch